jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/startupchecks/CheckForDuplicatePaths.java

```diff
@@ -0,0 +1,159 @@
+import se.fortnox.reactivewizard.jaxrs.JaxRsResource;
+import se.fortnox.reactivewizard.util.ReflectionUtil;
+
+import javax.inject.Inject;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CheckForDuplicatePaths implements StartupCheck {
+
+    private final        StartupCheckConfig startupCheckConfig;
+    private static final Logger             LOG                = LoggerFactory.getLogger(CheckForDuplicatePaths.class);
+    public static final  Pattern            PATH_PARAM_PATTERN = Pattern.compile("\\{(\\w*)\\}");
+
+    @Inject
+    public CheckForDuplicatePaths(StartupCheckConfig startupCheckConfig) {
+        this.startupCheckConfig = startupCheckConfig;
+    }
+
+    @Override
+    public void check(List<JaxRsResource> resources) {
+        resources.forEach(jaxRsResource -> {
+            final Optional<JaxRsResource> duplicate = resources.stream()
+                .filter(res -> res != jaxRsResource && pathParamCollides(res, jaxRsResource))
+                .findFirst();
+
+            if (duplicate.isPresent()) {
+                String errorMessage = String.format("%s duplicates %s on path %s",
+                    duplicate.get().getResourceMethod(),
+                    jaxRsResource.getResourceMethod(),
+                    duplicate.get().getPath());
+                if (startupCheckConfig.isFailOnError()) {
+                    throw new IllegalStateException(errorMessage);
+                } else {
+                    LOG.warn(errorMessage);
+                }
+            }
+        });
+    }
+
+    public static boolean pathParamCollides(JaxRsResource jaxRsResource1, JaxRsResource jaxRsResource2) {
+        if (jaxRsResource2 == null) {
+            return false;
+        }
+
+        final Matcher thisMatcher = PATH_PARAM_PATTERN.matcher(jaxRsResource1.getPath());
+        final Matcher thatMatcher = PATH_PARAM_PATTERN.matcher(jaxRsResource2.getPath());
+
+        boolean pathParamCollision = false;
+        while (thisMatcher.find() && thatMatcher.find()) {
+            final Type thisPathParamType = getTypeOfParam(thisMatcher.group(1), jaxRsResource1.getResourceMethod());
+            final Type thatPathParamType = getTypeOfParam(thatMatcher.group(1), jaxRsResource2.getResourceMethod());
+
+            if (!Objects.equals(thisPathParamType, thatPathParamType)) {
+                pathParamCollision = true;
+            }
+        }
+
+        //Different verbs causes no collision
+        if (!jaxRsResource1.getHttpMethod().equals(jaxRsResource2.getHttpMethod())) {
```

Comment:
This check is very cheap, moving it up would save some CPU from running the code above with the matchers.